odl/discr/discr_mappings.py

```diff
@@ -501,61 +445,66 @@ def __init__(self, ip_fspace, grid, dspace, order='C'):
-            The array of numbers to be interpolated
+        x : `FnBaseVector`
+            The array of values to be interpolated
+        out : `FunctionSpaceVector`, optional
+            Vector in which to store the interpolator
 
         Returns
         -------
-        outp : `FunctionSetVector`
-            A function (nearest-neighbor) interpolating at a given
-            point or array of points.
-
-        Examples
-        --------
-        Let's define the complex function space :math:`L^2` on a
-        rectangle:
-
-        TODO: implement an example!
-
+        out : `FunctionSpaceVector`
+            Linear interpolator for the grid of this
+            operator. Identical to input ``out`` if provided.
         """
-        def func(*x):
-            x = np.atleast_1d(x).squeeze()
-            values = interpnd(points=self.grid.coord_vectors,
-                              values=inp.data.reshape(self.grid.shape,
-                                                      order=self.order),
-                              method='nearest',
-                              xi=x,
-                              fill_value=None)  # Allow points outside
-            return values[0] if values.shape == (1,) else values
+        def linear(arg, out=None):
+            """Interpolating function with vectorization."""
+            if is_valid_input_meshgrid(arg, self.grid.ndim):
+                # TODO: check if this works for 'F' ordering
+                interp = _LinearMeshgridInterpolator(
+                    self.grid.coord_vectors,
+                    x.data.reshape(self.grid.shape, order=self.order))
+            else:
+                # TODO: check if points are ordered. If not, use
+                # LinearNDInterpolator
+                interp = _LinearPointwiseInterpolator(
+                    self.grid.coord_vectors,
+                    x.data.reshape(self.grid.shape, order=self.order))
+            return interp(arg, out=out)
 
-        return self.range.element(func)
+        return self.range.element(linear, vectorized=True)
 
 
-class _NearestPointwiseInterpolator(object):
+class _PointwiseInterpolator(object):
 
-    """Own version of SciPy's grid interpolator by point.
+    """Abstract interpolator class for pointwise interpolation.
 
-    We want to support non-numerical values for nearest neighbor
-    interpolation and in-place evaluation.
+    The code is adapted from SciPy's `RegularGridInterpolator
+    <http://docs.scipy.org/doc/scipy/reference/generated/\
+scipy.interpolate.RegularGridInterpolator.html>`_ class.
     """
 
     def __init__(self, coord_vecs, values):
         """Initialize a new instance."""
 
         # Provide values for some attributes
-        self.method = 'nearest'
         self.bounds_error = False
-        self.fill_value = None
+        # TODO: use fill_value depending on discretization
+        self.fill_value = None  # extrapolate
 
         if not hasattr(values, 'ndim'):
```

Comment:
It doesn't fail, it returns `False`.

This is unchanged from [the original code](https://github.com/scipy/scipy/blob/v0.16.1/scipy/interpolate/interpolate.py#L1630-L1632).